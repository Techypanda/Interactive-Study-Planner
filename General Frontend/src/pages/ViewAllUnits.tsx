import { useEffect, useState } from 'react';
import { Box, Button, Container,Typography, Card, CardContent, TextField} from "@material-ui/core";
import { useHistory } from "react-router-dom";
import axios from "axios";
import { DefaultProps } from "../types";
import { createStyles, makeStyles, Theme} from '@material-ui/core/styles';
import UnitsDataTable from "../components/shared/UnitsDatatable"

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
	root: {
	    display: 'flex',
	},
	details: {
	    display: 'flex',
	    flexDirection: 'column',
	},
	content: {
	    flex: '1 0 auto',
	},
	cover: {
	    width: 151,
	},
	controls: {
	    display: 'flex',
	    alignItems: 'right',
	    paddingRight: theme.spacing(1),
	    paddingBottom: theme.spacing(1),
	},
	playIcon: {
	    height: 38,
	    width: 38,
	},
	
    }),
);

function ViewAllUnits(props: DefaultProps) {
    const history = useHistory();
    const classes = useStyles();

    const base = [{Credits: "", Antirequisites: "", Prerequisites: "", Delivery: "", UnitCode: "", Description: "", Corequisites: "", Name: ""}];
    const [units_list, set_units_list] = useState(base);

    const GetAllUnits = () => {
	axios.get("https://ilur318q9c.execute-api.ap-southeast-2.amazonaws.com/Prod/getallunits/")
	.then((response) => {
	    set_units_list(response.data);
	});
    }
    useEffect(() => {
	GetAllUnits();
    }, []);

    return (
	<>
		<br/>
	    <Container  >
			<UnitsDataTable items={units_list}/>
		</Container>
	    <br/>

	</>
    );
}

export default ViewAllUnits;
