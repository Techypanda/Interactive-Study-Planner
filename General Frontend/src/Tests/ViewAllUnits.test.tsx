import React from 'react';
import { render } from '@testing-library/react';
import ViewAllUnits from '../pages/ViewAllUnits';

describe('View All Units is working when', () => {
    it('Renders without crashing', () => {
	render(<ViewAllUnits/>)
    })
    it('Renders and matches snapshot', () => {
	const x = render(<ViewAllUnits/>)
	expect(x).toMatchSnapshot();
    })
    it('Renders corresponding unit info pages', () => {
	jest.mock('react-router-dom', () => {
	    useHistory: () => {
		push: jest.fn();
	    }
	})
    })
})
