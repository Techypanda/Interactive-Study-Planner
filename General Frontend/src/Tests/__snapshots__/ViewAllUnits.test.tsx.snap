// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View All Units is working when Renders and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-18"
          id="searchcontainer"
        >
          <div
            class="MuiFormControl-root MuiTextField-root searchbar MuiFormControl-fullWidth"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="standard-full-width"
                placeholder="Search units..."
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-19 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-20"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained searchbtn"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Search
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <br />
        <div
          class="card-body"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-12 MuiPaper-outlined MuiPaper-rounded"
          >
            <div
              class="makeStyles-details-13"
            >
              <div
                class="MuiCardContent-root makeStyles-content-14"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
                />
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
                />
              </div>
              <div
                class="makeStyles-controls-16"
              />
            </div>
          </div>
        </div>
      </div>
      <br />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Back
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-18"
        id="searchcontainer"
      >
        <div
          class="MuiFormControl-root MuiTextField-root searchbar MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="standard-full-width"
              placeholder="Search units..."
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-19 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-20"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained searchbtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Search
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <br />
      <div
        class="card-body"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-root-12 MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            class="makeStyles-details-13"
          >
            <div
              class="MuiCardContent-root makeStyles-content-14"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
              />
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
              />
            </div>
            <div
              class="makeStyles-controls-16"
            />
          </div>
        </div>
      </div>
    </div>
    <br />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
